src/vaadin-grid-event-context-mixin.html

```diff
@@ -0,0 +1,60 @@
+<!--
+@license
+Copyright (c) 2018 Vaadin Ltd.
+This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+-->
+
+<script>
+  window.Vaadin = window.Vaadin || {};
+  window.Vaadin.Grid = window.Vaadin.Grid || {};
+
+  /**
+   * @polymerMixin
+   */
+  Vaadin.Grid.EventContextMixin = superClass => class EventContextMixin extends superClass {
+
+    /**
+     * Returns an object with context information about the event target:
+     * - `item`: the data object corresponding to the targeted row (not specified when targeting header or footer)
+     * - `column`: the column element corresponding to the targeted cell (not specified when targeting row details)
+     * - `section`: whether the event targeted the body, header, footer or details of the grid
+     *
+     * These additional properties are included when `item` is specified:
+     * - `index`: the index of the item
+     * - `selected`: the selected state of the item
+     * - `detailsOpened`: whether the row details are open for the item
+     * - `expanded`: the expanded state of the tree toggle
+     * - `level`: the tree hierarchy level represented with a horizontal offset of the toggle button
+     *
+     * The returned object is populated only when a grid cell, header, footer or row details is found in `event.composedPath()`.
+     * This means mostly mouse and keyboard events. If such a grid part is not found in the path, an empty object is returned.
+     * This may be the case eg. if the event is fired on the `<vaadin-grid>` element and not any deeper in the DOM, or if
+     * the event targets the empty part of the grid body.
+     */
+    getEventContext(event) {
+
+      const context = {};
+
+      const path = event.composedPath();
+      const cell = path[path.indexOf(this.$.table) - 3];
```

Comment:
this part of logic is already present in `ActiveItemMixin`, could we extract it into internal method, e.g. `_getCellByEvent(event)`?

Reply:
In that case `vaadin-grid.html` would be the only common place. If there isn't that much shared code, would prefer not to bloat the main class.

Reply:
Sounds logical. We can keep at as it is for now.